js/repl/src/Repl.js

```diff
@@ -0,0 +1,413 @@
+    };
+
+    // Load any plug-ins enabled by query params
+    this._checkForUnloadedPlugins();
+  }
+
+  render() {
+    const state = this.state;
+
+    const options = {
+      lineWrapping: state.lineWrap
+    };
+
+    return (
+      <div className={styles.repl}>
+        <ReplOptions
+          builtIns={state.builtIns}
+          className={styles.optionsColumn}
+          debugEnvPreset={state.debugEnvPreset}
+          envConfig={state.envConfig}
+          envPresetState={state.envPresetState}
+          isExpanded={state.isSidebarExpanded}
+          lineWrap={state.lineWrap}
+          pluginState={state.plugins}
+          presetState={state.presets}
+          runtimePolyfillConfig={runtimePolyfillConfig}
+          runtimePolyfillState={state.runtimePolyfillState}
+          toggleEnvPresetSetting={this._toggleEnvPresetSetting}
+          toggleIsExpanded={this._toggleIsSidebarExpanded}
+          toggleSetting={this._toggleSetting}
+        />
+
+        <div className={styles.panels}>
+          <CodeMirrorPanel
+            className={styles.codeMirrorPanel}
+            code={state.code}
+            error={state.compileError}
+            onChange={this._updateCode}
+            options={options}
+            placeholder="Write code here"
+          />
+          <CodeMirrorPanel
+            className={styles.codeMirrorPanel}
+            code={state.compiled}
+            error={state.evalError}
+            info={state.debugEnvPreset ? state.envPresetDebugInfo : null}
+            options={options}
+          />
+        </div>
+      </div>
+    );
+  }
+
+  _checkForUnloadedPlugins() {
+    const {
+      envConfig,
+      envPresetState,
+      plugins,
+      runtimePolyfillState
+    } = this.state;
+
+    // Assume all default presets are baked into babel-standalone.
+    // We really only need to worry about plugins.
+    for (const key in plugins) {
+      const plugin = plugins[key];
+
+      if (plugin.isEnabled && !plugin.isLoaded && !plugin.isLoading) {
+        this._numLoadingPlugins++;
+
+        loadPlugin(plugin, success => {
+          this._numLoadingPlugins--;
+
+          if (!success) {
+            this.setState(state => ({
+              plugins
+            }));
+          }
+
+          // Once all plugins have been loaded, re-compile code.
+          if (this._numLoadingPlugins === 0) {
+            this._updateCode(this.state.code);
+          }
+        });
+      }
+    }
+
+    // Babel (runtime) polyfill is large;
+    // It's only needed if we're actually executing the compiled code.
+    // Defer loading it unless "evaluate" is enabled.
+    if (runtimePolyfillState.isEnabled && !runtimePolyfillState.isLoaded) {
+      loadPlugin(runtimePolyfillState, () => {
+        let evalError = null;
+
+        // No need to recompile at this point;
+        // Just evaluate the most recently compiled code.
+        try {
+          // eslint-disable-next-line
+          eval(this.state.compiled);
```

Comment:
We'll only hit these lines of "evaluate" is checked. (It toggles `runtimePolyfillState.isEnabled`. The comments a few lines up mention this.)